### PR TITLE
Css auto again

### PR DESCRIPTION
**Pull request by maho**
I'm kind-of-re-opening pull request at https://github.com/pyjs/pyjs/pull/761

You said, that you agree but you think that it's no way to go and let's start discussion. I started (tried to): https://groups.google.com/forum/?fromgroups=#!searchin/pyjs-users/autocss/pyjs-users/JKJW8MVqipQ/axEjoNcM6h8J - nobody interested in discussing. 

However, I think that change is small enough, but very useful. I agree, embedding etc.. but so far we're not embedding but we use .css anyways - so why not facilitate it?

### CODE REVIEW

**Comment by anthonyrisinger** [comment]
hmm, i'll think this one over a bit... it's certainly a problem, but there are some potential side effects i really don't like (2x HTTP reqs, no way to opt-out, no way to "unset", CSS loaded _after_ DOM [flash brokenness], etc)... will explore solutions.

**Comment by maho** [comment]
W dniu 07.12.2012 07:15, C Anthony Risinger pisze:

> hmm, i'll think this one over a bit... it's certainly a problem, but
> there are some potential side effects i really don't like (2x HTTP reqs,
> no way to opt-out, no way to "unset", CSS loaded /after/ DOM [flash
> brokenness], etc)... will explore solutions.

I think it's not 2x http reqs, because you have .css for widgets 
anyways, but you need to include them "by hand" to application/html.

Yes, right, you usually merge them by hand into one .css. And I think 
it's matter of another tasks - merge .css files used in project by 
pyjsbuilder.

But - I still insist that CSSauto is better than nothing. If I do some 
"opt out" mechanism, will you accept it?

BTW - is there any way to "hook" at compile time? Eg. define part of 
code in file which is executed in compile time, which would generate 
merged .css file? It's subject of another task/branch I think...

## 

pozdrawiam

Łukasz Mach - lukasz.mach@pagema.net

File: library/pyjamas/ui/UIObject.py [hunk 1]
```diff
@@ -188,6 +191,18 @@ def removeStyleDependentName(self, styleSuffix):
         """
         self.removeStyleName(self.getStylePrimaryName()+"-"+styleSuffix)
 
+
```
Comment (maho) [inline library/pyjamas/ui/UIObject.py:194]: I don't know what to do with this method. Remove, or leave (maybe it will be useful in future?)

**Comment by maho** [comment]
After fruitful discussion at malinglist, I modified it, to totally different approach, I believe - acceptable by you. 

Please check. 

There is also method in UIObject which is no longer needed in this branch, but maybe it will be useful in future? I don't know if remove it or not.

**Comment by maho** [comment]
I'm curious - if you didn't look into change yet, or it's not acceptable for some reason?

**Comment by anthonyrisinger** [comment]
bleh i forgot to reference this request, but i merged the essence of your work in `d36e81fe2f`,  `fa92244293`, and `4279d5a22d`. i mostly merged/split with a few style changes (spaces are nice on the eyes! :) and moved the impl from `linker.py` to `browser.py` (the browser specific linker) in addition to merging your function with `merge_resources`.

i probably mentioned this on list at some point, but i'm somewhat offline ATM due to my $DAYJOB (tax-related... company i do work for makes all their income for the year in the next 6-12 weeks or so)... so i've been working late and weekends often :( but when this settles i'll be more on the ball.

that said, if you have continuing changes, i don't have much issue with granting you commit access providing you simply continue to ping the list and also create merge requests (then merge them yourself after any input).  i would ask that you try to remedy a few things i've noticed as pattern:
- don't add/remove unrelated extra whitespace (git diff!)
- use 4 spaces, not tabs
- use **more** spaces! mainly in function args/etc (generally PEP8 as appropriate)
- your editor appears to insert a UTF-8 BOM mark in files you **create**, need to make it stop doing that! (windows is notoriously bad about this, but it can cause big probs for everyone else, and BOM mark is all but dead [and optional anyway])
- try to split commits by function/purpose, but ensure each one is "complete" (eg. pyjs is not "broken" afterwards)

...and learn some `git foo-magic`! it will serve you well in long run methinks, here and elsewhere :)

as always, great work... thx much

**Comment by maho** [comment]
Well, I think that I avoided problems with whitespace, after your previous notices, seems that I didn't. BTW - I reviewed diff of this branch, and I don't see any unrelated extra whitespaces or lack of whitespaces in functions. 

Regarding BOM and 4spaces - I will check more carefully. I use Eclipse, I thought that it's quite friendly to PEP8.

**Comment by anthonyrisinger** [comment]
for new additions you can generally format as best you can, with some discretion as to not diverge significantly from surrounding style (if any).

i think my previous comments (in past commits) related to _mixing_ whitespace changes along with the introduction of new functionality.  IME, consistency, even if [tolerably] against best-practices, yields far fewer surprises, and allows for speedy review and simpler refactoring later on. alas, i know i'm pretty particular about my own writings, so i try my best not to impose my own practices too hard :)

your code was perfectly fine, i mainly just wanted to move it to `browser.py` and i simply made some tweaks once i got started.

tl;dr... you've done fine work, here and in past, and it's very much appreciated. no worries at all!
